src/repr/src/bytes.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2018 sqlparser-rs contributors. All rights reserved.
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// This file is derived from the sqlparser-rs project, available at
+// https://github.com/andygrove/sqlparser-rs. It was incorporated
+// directly into Materialize on December 21, 2019.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use mz_ore::cast::CastLossy;
+
+/// Struct for postgres compatible size units which is slightly different from
+/// [`bytesize::ByteSize`]. Instead of MiB or GiB and so on, it uses MB, GB for the sizes
+/// with 1024 multiplier.
+/// Valid units are B, KB, MB, GB, TB, PB with multiples of 1024
+/// where 1MB = 1024 KB
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize, Default)]
+pub struct Bytes(pub u64);
```

Comment:
This name seemed more appropriate, happy to take other suggestions too.

Reply:
Personally I would prefer `ByteSize` or `ByteAmount` since [`Bytes`](https://crates.io/crates/bytes) is a commonly used struct that is a container for blobs of bytes